packages/presets/client/src/typescript-ast-visitor/typescript-variables-to-object.ts

```diff
@@ -0,0 +1,92 @@
+// TODO: get rid of
+import {
+  AvoidOptionalsConfig,
+  ConvertNameFn,
+  normalizeAvoidOptionals,
+  NormalizedScalarsMap,
+  OperationVariablesToObject,
+  ParsedDirectiveArgumentAndInputFieldMappings,
+  ParsedEnumValuesMap,
+} from '@graphql-codegen/visitor-plugin-common';
+import { Kind, TypeNode } from 'graphql';
+
+export class TypeScriptOperationVariablesToObject extends OperationVariablesToObject {
+  constructor(
+    _scalars: NormalizedScalarsMap,
+    _convertName: ConvertNameFn,
+    private _avoidOptionals: boolean | AvoidOptionalsConfig,
+    private _immutableTypes: boolean,
+    _namespacedImportName: string | null = null,
+    _enumNames: string[] = [],
+    _enumPrefix = true,
+    _enumValues: ParsedEnumValuesMap = {},
+    _applyCoercion: Boolean = false,
+    _directiveArgumentAndInputFieldMappings: ParsedDirectiveArgumentAndInputFieldMappings = {},
```

Comment:
why do we prefix arguments with `_` ? also this constructor accept too many arguments, consider switch to object as one argument

Reply:
this is just moved from another place, and in progress to refractor and eventually get rid of